plugins/cloudinary/src/CloudinaryImageEditor.tsx

```diff
@@ -0,0 +1,162 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import React from 'react'
+import { Builder } from '@builder.io/sdk'
+import { Typography, Button } from '@material-ui/core'
+import {
+  CloudinaryMediaLibraryDialog,
+  CloudinaryImage
+} from './CloudinaryMediaLibraryDialog'
+import CloudinayCredentialsDialog from './CloudinaryCredentialsDialog'
+
+interface Props {
+  value?: string
+  onChange(newValue: CloudinaryImage): void
+  context: any
+}
+
+// No need to use username for cloudinary login if SSO is enabled
+interface CloudinaryImageEditorState {
+  showDialog: boolean
+  apiKey: string | undefined
+  cloudName: string | undefined
+  requestCredentials: boolean
+}
+
+export default class CloudinaryImageEditor extends React.Component<
+  Props,
+  CloudinaryImageEditorState
+> {
+  get organization() {
+    return this.props.context.user.organization
+  }
+
+  get cloudinaryCloud(): string | undefined {
+    return this.organization.value.settings.plugins.get('cloudinaryCloud')
+  }
+
+  set cloudinaryCloud(cloud: string | undefined) {
+    this.organization.value.settings.plugins.set('cloudinaryCloud', cloud)
+    this.organization.save()
+  }
+
+  get cloudinaryKey(): string | undefined {
+    return this.organization.value.settings.plugins.get('cloudinaryKey')
+  }
+
+  set cloudinaryKey(key: string | undefined) {
+    this.organization.value.settings.plugins.set('cloudinaryKey', key)
+    this.organization.save()
+  }
+
+  constructor(props: any) {
+    super(props)
+    this.state = {
+      requestCredentials: false,
+      showDialog: false,
+      apiKey: this.cloudinaryKey ? this.cloudinaryKey : '',
+      cloudName: this.cloudinaryCloud ? this.cloudinaryCloud : ''
+    }
+  }
+
+  private closeDialog() {
+    this.setState({
+      requestCredentials: false,
+      showDialog: false
+    })
+  }
+
+  private appendMediaLibraryScriptToPlugin() {
+    const script = document.createElement('script')
+    script.async = true
+    script.src = 'https://media-library.cloudinary.com/global/all.js'
+    document.head.appendChild(script)
+  }
+
+  private shouldRequestCloudinaryCredentials() {
+    return (
+      this.state.requestCredentials ||
+      this.state.apiKey === '' ||
+      this.state.cloudName === ''
+    )
+  }
+
+  private updateCloudinaryCredentials(apiKey: string, cloudName: string) {
+    this.cloudinaryKey = apiKey
+    this.cloudinaryCloud = cloudName
+    this.setState({
+      apiKey: this.cloudinaryKey,
+      cloudName: this.cloudinaryCloud
+    })
+  }
+
+  private selectImage(image: CloudinaryImage) {
+    this.props.onChange(image)
+  }
+
+  componentDidMount() {
+    this.appendMediaLibraryScriptToPlugin()
+  }
+
+  render() {
+    const shouldRequestCloudinarySettings = this.shouldRequestCloudinaryCredentials()
+    const buttonStyle = { width: '45%', margin: '5px' }
```

Comment:
This is an arbritrary UI decision... 2 buttons with same width within the same row. One to pick up the image, one to (re)set the Cloudinary credentials.